### PR TITLE
Added and configured husky and lint-staged.

### PR DESCRIPTION
**Pull request by nicolad**
## Problem this Pull Request solves
Implements: https://github.com/eventespresso/event-espresso-core/issues/1784
This setup helps us run some specific checks before commit is being made. If checks are failing commit is not completed.

## How has this been tested
Peek 2019-11-07 14-27

## Checklist

* [x] I have read the documentation relating to systems affected by this pull request, see https://github.com/eventespresso/event-espresso-core/tree/master/docs
* [ ] User input is adequately validated and sanitized
* [ ] all publicly displayed strings are internationalized (usually using `esc_html__()`, see https://codex.wordpress.org/I18n_for_WordPress_Developers)
* [ ] My code is tested.
* [ ] My code follows the Event Espresso code style.
* [ ] My code has proper inline documentation.
* [ ] My code accounts for when the site is in Maintenance Mode (MM2 especially disallows usage of models)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,33 +16,43 @@
 		}
 	},
 	"scripts": {
```
Comment (nicolad) [inline package.json]: Here I've ordered scripts alphabetically.

Comment (nicolad) [reply package.json]: And added:
```
"lint-js": "wp-scripts lint-js",
"lint-js:fix": "npm run lint-js -- --fix",
```

Comment (manzoorwanijk) [reply package.json]: Yes, got that... It's good to be sorted 👍

File: package.json [hunk 2]
```diff
@@ -1,6 +1,8 @@
 {
 	"name": "@eventespresso/core",
 	"version": "1.0.0",
+	"author": "Event Espresso",
+	"license": "GPL-2.0",
```
Comment (tn3rb) [inline package.json]: we're bumping up to GPLv3 now

File: package.json [hunk 3]
```diff
@@ -14,33 +16,43 @@
...
 		"test-unit": "wp-scripts test-unit-js --config tests/javascript-config/unit/jest.config.json",
-		"test-unit:update": "npm run test-unit -- --updateSnapshot",
-		"test-unit:watch": "npm run test-unit -- --watch",
 		"test-unit:coverage": "npm run test-unit -- --coverage",
 		"test-unit:coverage-ci": "npm run test-unit -- --coverage --maxWorkers 1 && codecov",
-		"lc": "jsgl --local ./",
-		"build:css": "node ./bin/css-builder/css-builder.js"
+		"test-unit:update": "npm run test-unit -- --updateSnapshot",
+		"test-unit:watch": "npm run test-unit -- --watch",
+		"watch": "concurrently \"webpack --watch --config webpack.dev.js\" \"npm-watch build:css\"",
```
Comment (tn3rb) [inline package.json]: hey I've been thinking it would be handy to have a command for running a dev build but without the watch flag added.
ie: something like:

```js
		"dev": "concurrently \"webpack --config webpack.dev.js\" \"npm-watch build:css\"",
```

is there already a convention for naming a command like that other than "dev" ?

If so, we can use that...
although imho it would make more sense to have:

- `npm run dev`
- `npm run prod` 

rather than

- `npm run watch`
- `npm run build`

and for the watch and polling dev builds we could have:

- `npm run dev-w` for `webpack --watch --config webpack.dev.js`
- `npm run dev-p` for `webpack --watch --watch-poll --config webpack.dev.js`

what do you think?
I have no experience with these things outside of this project, so if my suggestions are comical, then plz feel free to laugh! I can take it !

Bottom line is that we should follow standard conventions as much as possible for the benefit of all

Comment (nicolad) [reply package.json]: In context of gatsby projects I've seen `dev` being used over the longer `develop`, so I am fine with `dev`, rather than `watch` or `develop`.
I've noticed `build` being more often used than `prod`, but don't want to debate over this, but rather find an agreement and move forward since this is a minor thing.

File: package.json [hunk 4]
```diff
@@ -22,6 +22,9 @@
 		"ci:build": "npm run build",
 		"ci:lint": "npm run lint",
 		"ci:test": "npm run test-unit:coverage-ci",
+		"dev": "concurrently \"webpack --config webpack.dev.js\" \"npm-watch build:css\"",
+		"dev-w": "webpack --watch --config webpack.dev.js",
```
Comment (tn3rb) [inline package.json]: this would still need to be 

```suggestion
		"dev-w": "concurrently \"webpack --watch --config webpack.dev.js\" \"npm-watch build:css\"",
```

in order to build CSS as well no?

Comment (nicolad) [reply package.json]: I saw that it's different than the old `watch`, but I trusted and copy-pasted that from you :D

Comment (tn3rb) [reply package.json]: HA!!

ya don't blindly follow **MY** suggestions!!!

File: js-green-licenses.json [hunk 5]
```diff
@@ -4,6 +4,7 @@
 		"GPL-2.0",
 		"GPL-2.0-or-later",
 		"GPL-2.0+",
+		"GPL-3.0",
```
Comment (nicolad) [inline js-green-licenses.json]: Interesting, I thought that `GPL-3.0` is included in `"GPL-2.0-or-later"`, in the `or-later` part :)

Comment (tn3rb) [reply js-green-licenses.json]: ya I thought the exact same thing, but 2+ must mean things like 2.1, 2.1, etc
